### PR TITLE
FarhanNurzi/P2PS-1107/Refactor CancelAddPaymentMethodModal

### PR DESCRIPTION
**Pull request by farhan-nurzi-deriv**
## Changes:

Please provide a summary of the change.

### Screenshots:
Screenshot 2023-08-21 at 10 54 02 AM

### CODE REVIEW

File: packages/p2p/src/components/modal-manager/modals/cancel-add-payment-method-modal/__tests__/cancel-add-payment-method-modal.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        render(<CancelAddPaymentMethodModal />);
+        expect(screen.getByText('Cancel adding this payment method?')).toBeInTheDocument();
+        expect(
+            screen.getByText('If you choose to cancel, the details you’ve entered will be lost.')
+        ).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: 'Cancel' })).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: 'Go back' })).toBeInTheDocument();
+    });
+
+    it('should cancel adding payment method', () => {
```
Comment (niloofar-deriv) [inline packages/p2p/src/components/modal-manager/modals/cancel-add-payment-method-modal/__tests__/cancel-add-payment-method-modal.spec.tsx]: Can we write a better description for this test?

File: packages/p2p/src/components/modal-manager/modals/cancel-add-payment-method-modal/__tests__/cancel-add-payment-method-modal.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        const mock_on_cancel = jest.fn();
+        render(<CancelAddPaymentMethodModal onCancel={mock_on_cancel} should_hide_all_modals_on_cancel />);
+        const cancel_button = screen.getByRole('button', { name: 'Cancel' });
+        userEvent.click(cancel_button);
+
```
Comment (niloofar-deriv) [inline packages/p2p/src/components/modal-manager/modals/cancel-add-payment-method-modal/__tests__/cancel-add-payment-method-modal.spec.tsx:59]:
```suggestion
const { hideAddPaymentMethodForm, setSelectedPaymentMethod  } = mock_stores.my_profile_store
```

File: packages/p2p/src/components/modal-manager/modals/cancel-add-payment-method-modal/__tests__/cancel-add-payment-method-modal.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        expect(mock_stores.my_profile_store.hideAddPaymentMethodForm).toHaveBeenCalled();
```
Comment (niloofar-deriv) [inline packages/p2p/src/components/modal-manager/modals/cancel-add-payment-method-modal/__tests__/cancel-add-payment-method-modal.spec.tsx]:
```suggestion
        expect(hideAddPaymentMethodForm).toHaveBeenCalled();
```

File: packages/p2p/src/components/modal-manager/modals/cancel-add-payment-method-modal/__tests__/cancel-add-payment-method-modal.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        expect(mock_stores.my_profile_store.setSelectedPaymentMethod).toHaveBeenCalledWith('');
```
Comment (niloofar-deriv) [inline packages/p2p/src/components/modal-manager/modals/cancel-add-payment-method-modal/__tests__/cancel-add-payment-method-modal.spec.tsx]:
```suggestion
        expect(setSelectedPaymentMethod).toHaveBeenCalledWith('');
```

File: packages/p2p/src/components/modal-manager/modals/cancel-add-payment-method-modal/__tests__/cancel-add-payment-method-modal.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        const cancel_button = screen.getByRole('button', { name: 'Cancel' });
+        userEvent.click(cancel_button);
+
+        expect(mock_modal_manager_context.hideModal).toHaveBeenCalledWith({
+            should_save_form_history: false,
+            should_hide_all_modals: false,
+        });
+    });
+
+    it('should hide modal when clicking go back button', () => {
```
Comment (niloofar-deriv) [inline packages/p2p/src/components/modal-manager/modals/cancel-add-payment-method-modal/__tests__/cancel-add-payment-method-modal.spec.tsx]:
```suggestion
    it('should hide modal when clicking on go back button', () => {
```

**Review by ameerul-deriv — APPROVED** [review]

**Review by niloofar-deriv — APPROVED** [review]

**Review by farrah-deriv — APPROVED** [review]

**Pull request merged**